skbio/io/format/genbank.py

```diff
@@ -0,0 +1,468 @@
+        feature, pmd = _parse_single_feature(section, length, i)
+        features.append(feature)
+        positional_metadata.append(pmd)
+    return features, positional_metadata
+
+
+def _parse_single_feature(lines, length, index):
+    '''Parse a feature.
+
+    Returns
+    -------
+    tuple
+        Tuple of a dict of `metadata` and a pandas.Series of
+        `positional_metadata` for the feature.
+
+    '''
+    feature = dict()
+    feature['index_'] = index
+    # each component of a feature starts with '/', except the 1st
+    # component of location.
+    section_splitter = yield_section(
+        lambda x: x.startswith('/'), strip=True)
+    first = True
+    for section in section_splitter(lines):
+        if first:
+            # first section is the Location string
+            first = False
+            type, location = _parse_section_default(
+                section, join_delimitor='', return_label=True)
+            feature['type_'] = type
+            feature['location'] = location
+            loc, loc_pmd = _parse_loc_str(location, length)
+            feature.update(loc)
+        else:
+            # following sections are Qualifiers
+            k, v = _parse_section_default(
+                section, label_delimitor='=',
+                join_delimitor=' ', return_label=True)
+            k = k[1:]
+            # strip the quotes if it is quoted.
+            # v could be empty
+            if v and v[0] == '"' and v[-1] == '"':
+                v = v[1:-1]
+            # some Qualifiers can appear multiple times
+            if k in feature:
+                feature[k] = [feature[k]]
+                feature[k].append(v)
+            else:
+                feature[k] = v
+    return feature, loc_pmd
+
+
+def _parse_loc_str(loc_str, length):
+    '''Parse location string.
+
+    Warning: This converts coordinates to 0-based from 1-based as
+    in Genbank format.
+    '''
+    pmd = np.zeros(length, dtype=bool)
+    res = {'rc_': False,
+           'left_partial_': False,
+           'right_partial_': False}
+    items = re.split('[(),]+', loc_str)
+    operators = ['join', 'complement']
+    if 'complement' in items:
+        res['rc_'] = True
+    for i in items:
+        i = i.strip()
+        if i in operators or not i:
+            continue
+        elif '..' in i:  # span
+            beg, end = i.split('..')
+            if beg.startswith('<'):
+                beg = beg[1:]
+                res['left_partial_'] = True
+            if end.startswith('>'):
+                end = end[1:]
+                res['right_partial_'] = True
+            beg = int(beg)
+            end = int(end)
+            index = range(beg-1, end)
+        elif i.isdigit():  # single base
+            index = int(i) - 1
+        else:
+            raise GenbankFormatError(
+                'Could not parse location string:\n%s' %
+                loc_str)
+        pmd[index] = True
+
+    return res, pd.Series(pmd)
```

Comment:
> Maybe we should leave it dense for now and work on improving pandas upstream?

Sounds like a good plan. Can one of you please create an issue on our tracker to investigate this further?